File: MasiWasiMovil/app/src/main/java/com/example/masiwasimovil/activities/DetailActivity.java
package com.example.masiwasimovil.activities;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.bumptech.glide.Glide;
import com.example.masiwasimovil.R;

public class DetailActivity extends AppCompatActivity {

    private ImageView fotoMascota;
    private TextView nombreMascota, sexo, edad, categoria, color, descripcion;
    private Button contactar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        // 1. Inicializar vistas
        fotoMascota = findViewById(R.id.fotoMascota);
        nombreMascota = findViewById(R.id.nombreMascota);
        sexo = findViewById(R.id.sexo);
        edad = findViewById(R.id.edad);
        categoria = findViewById(R.id.categoria);
        color = findViewById(R.id.color);
        descripcion = findViewById(R.id.descripcion);
        contactar = findViewById(R.id.contactar);

        // 2. Obtener los datos del Intent (deben coincidir con las llaves de HomeFragment)
        String nombre = getIntent().getStringExtra("nombre");
        String desc = getIntent().getStringExtra("descripcion");
        String url = getIntent().getStringExtra("imagenUrl");

        // Si pasaste más datos, recíbelos aquí también:
        String sexoTxt = getIntent().getStringExtra("sexo");
        String edadTxt = getIntent().getStringExtra("edad");
        String catTxt = getIntent().getStringExtra("categoria");
        String colorTxt = getIntent().getStringExtra("color");

        // 3. Validar y mostrar
        if (nombre == null) {
            Toast.makeText(this, "Error al cargar datos", Toast.LENGTH_SHORT).show();
            finish();
            return;
        }

        llenarPantalla(nombre, desc, url, sexoTxt, edadTxt, catTxt, colorTxt);

        // 4. Configurar botón contactar
        contactar.setOnClickListener(v -> {
            Toast.makeText(this, "Redirigiendo al perfil del dueño...", Toast.LENGTH_SHORT).show();
            Intent intent = new Intent(DetailActivity.this, MainActivity.class);
            intent.putExtra("navigateTo", "profile");
            startActivity(intent);
            finish();
        });
    }

    private void llenarPantalla(String n, String d, String u, String s, String e, String ct, String cl) {
        nombreMascota.setText(n);
        descripcion.setText(d != null ? d : "Sin descripción");

        // Llenar campos extra (con validación de nulos)
        sexo.setText("Sexo: " + (s != null ? s : "N/A"));
        edad.setText("Edad: " + (e != null ? e : "N/A"));
        categoria.setText("Categoría: " + (ct != null ? ct : "N/A"));
        color.setText("Color: " + (cl != null ? cl : "N/A"));

        // 5. CARGA DE IMAGEN REAL DESDE FIREBASE
        Glide.with(this)
                .load(u) // La URL que llega de Firestore
                .placeholder(R.drawable.ic_launcher_foreground) // Imagen temporal
                .error(R.mipmap.mascota1) // Si no hay foto, usa la de respaldo
                .into(fotoMascota);
    }
}

File: movil2/mozita/app/src/main/java/pe/edu/idat/mozitaapp/activities/DetailActivity.kt
package pe.edu.idat.mozitaapp.activities

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import pe.edu.idat.mozitaapp.R

class DetailActivity : AppCompatActivity() {

    private lateinit var fotoMascota: ImageView
    private lateinit var nombreMascota: TextView
    private lateinit var sexo: TextView
    private lateinit var edad: TextView
    private lateinit var categoria: TextView
    private lateinit var color: TextView
    private lateinit var descripcion: TextView
    private lateinit var contactar: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        // Vistas
        fotoMascota = findViewById(R.id.fotoMascota)
        nombreMascota = findViewById(R.id.nombreMascota)
        sexo = findViewById(R.id.sexo)
        edad = findViewById(R.id.edad)
        categoria = findViewById(R.id.categoria)
        color = findViewById(R.id.color)
        descripcion = findViewById(R.id.descripcion)
        contactar = findViewById(R.id.contactar)

        // Obtener datos del Intent
        val nombre = intent.getStringExtra("nombre")
        val desc = intent.getStringExtra("descripcion")
        val url = intent.getStringExtra("imagenUrl")

        val sexoTxt = intent.getStringExtra("sexo")
        val edadTxt = intent.getStringExtra("edad")
        val catTxt = intent.getStringExtra("categoria")
        val colorTxt = intent.getStringExtra("color")

        // Validar
        if (nombre == null) {
            Toast.makeText(this, "Error al cargar datos", Toast.LENGTH_SHORT).show()
            finish()
            return
        }

        llenarPantalla(nombre, desc, url, sexoTxt, edadTxt, catTxt, colorTxt)

        val duenoId = intent.getStringExtra("duenoId")

        // Botón contactar
        contactar.setOnClickListener {
            if (duenoId != null) {
                val intent = Intent(this, PersonActivity::class.java)
                // Pasamos el ID real para que PersonActivity haga la consulta
                intent.putExtra("DUENO_ID", duenoId)
                startActivity(intent)
            } else {
                Toast.makeText(this, "ID de dueño no disponible", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun llenarPantalla(
        n: String,
        d: String?,
        u: String?,
        s: String?,
        e: String?,
        ct: String?,
        cl: String?
    ) {

        nombreMascota.text = n
        descripcion.text = d ?: "Sin descripción"

        sexo.text = "Sexo: ${s ?: "N/A"}"
        edad.text = "Edad: ${e ?: "N/A"}"
        categoria.text = "Categoría: ${ct ?: "N/A"}"
        color.text = "Color: ${cl ?: "N/A"}"

        // Lógica de imagen local
        val categoriaTxt = ct?.lowercase() ?: "otros"

        val resourceId = when (categoriaTxt) {
            "perro" -> R.mipmap.perro
            "gato" -> R.mipmap.gato
            else -> R.mipmap.otros
        }

        // Se asigna el recurso directamente
        fotoMascota.setImageResource(resourceId)
    }
}